plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -285,14 +302,40 @@ object StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext<ResultTrack
         variableAssignment: FirVariableAssignment,
         data: StmtConversionContext<ResultTrackingContext>,
     ): Exp {
+
         // It is not entirely clear whether we can get away with ignoring the distinction between
         // lvalues and rvalues, but let's try to at first, and we'll fix it later if it turns out
         // not to work.
-        val convertedLValue = data.convert(variableAssignment.lValue)
-        val lValueType = data.embedType(variableAssignment.lValue)
-        val convertedRValue = data.convert(variableAssignment.rValue)
-        val rValueType = data.embedType(variableAssignment.rValue)
-        data.addStatement(Stmt.assign(convertedLValue, convertedRValue.convertType(rValueType, lValueType)))
+
+        if (variableAssignment.isAssigningToClassField) {
+
+            val lValue = variableAssignment.lValue as FirPropertyAccessExpression
+            val lValueSymbol = lValue.calleeSymbol as FirPropertySymbol
+            val lValueType = data.embedType(variableAssignment.lValue)
+            val receiver = data.convert(lValue.dispatchReceiver)
+
+
+            val varEmbedding = VariableEmbedding(lValueSymbol.callableId.embedName(), lValueType)
+            val fieldAccess = Exp.FieldAccess(receiver, varEmbedding.toField())
+
+            val convertedRValue = data.convert(variableAssignment.rValue)
+            val rValueType = data.embedType(variableAssignment.rValue)
+
+            data.withoutResult {
```

Comment:
By the way, what does `withoutResult` accomplish here?

Reply:
Just remove it because it didn't add value